src/EditorFeatures/Core/Shared/Utilities/SyntacticClassificationTypeMap.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.ComponentModel.Composition;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.VisualStudio.Text.Classification;
+
+namespace Microsoft.CodeAnalysis.Editor.Shared.Utilities
+{
+    [Export]
+    internal class SyntacticClassificationTypeMap : AbstractClassificationTypeMap
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public SyntacticClassificationTypeMap(IClassificationTypeRegistryService registryService)
+            : base(registryService, ClassificationLayer.Syntactic)
```

Comment:
After my [Editor change](https://devdiv.visualstudio.com/DevDiv/_git/VS-Platform/pullrequest/352931) this should have no impact on non-LSP colors, but it would be great if someone from Roslyn could validate a build with these two changes together, just in case.

Reply:
@dibarbet can you also take a look? I understand you have done a lot with Roslyn's classification tagging + LSP hybrid scenario.

Reply:
I think it is reasonable to expect either LSP semantic tokens or our semantic tagger to always override the syntactic classifications produced by our tagger.  This is mainly a curiosity question since we don't activate textmate, but what layer are textmate classifications at?

Reply:
TextMate is in the syntactic layer. As of [this change](https://devdiv.visualstudio.com/DevDiv/_git/VS-Platform/pullrequest/352931) the behavior is a bit more nuanced... layered colors are ordered relative to other layered colors but non-layered colors, like the ones in your semantic tagger, are MEF ordered.

Reply:
If the change works 100% as expected, passing `ClassificationLayer.Syntactic` should have no impact on mainline C# scenarios, but it _should_ cause your colors to 'duck out of the way' in LSP scenarios.